homeassistant/components/metoffice/__init__.py

```diff
@@ -51,91 +48,64 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     coordinates = f"{latitude}_{longitude}"
 
-    @callback
-    def update_unique_id(
-        entity_entry: er.RegistryEntry,
-    ) -> dict[str, Any] | None:
-        """Update unique ID of entity entry."""
-
-        if entity_entry.domain != Platform.SENSOR:
-            return None
-
-        name_to_key = {
-            "Station Name": "name",
-            "Weather": "weather",
-            "Temperature": "temperature",
-            "Feels Like Temperature": "feels_like_temperature",
-            "Wind Speed": "wind_speed",
-            "Wind Direction": "wind_direction",
-            "Wind Gust": "wind_gust",
-            "Visibility": "visibility",
-            "Visibility Distance": "visibility_distance",
-            "UV Index": "uv",
-            "Probability of Precipitation": "precipitation",
-            "Humidity": "humidity",
-        }
-
-        match = re.search(f"(?P<name>.*)_{coordinates}.*", entity_entry.unique_id)
-
-        if match is None:
-            return None
-
-        if (name := match.group("name")) in name_to_key:
-            return {
-                "new_unique_id": entity_entry.unique_id.replace(name, name_to_key[name])
-            }
-        return None
-
-    await er.async_migrate_entries(hass, entry.entry_id, update_unique_id)
-
-    connection = datapoint.connection(api_key=api_key)
-
-    site = await hass.async_add_executor_job(
-        fetch_site, connection, latitude, longitude
-    )
-    if site is None:
-        raise ConfigEntryNotReady
+    connection = datapoint.Manager.Manager(api_key=api_key)
 
-    async def async_update_3hourly() -> MetOfficeData:
+    async def async_update_daily() -> datapoint.Forecast:
         return await hass.async_add_executor_job(
-            fetch_data, connection, site, MODE_3HOURLY
+            fetch_data, connection, latitude, longitude, "daily"
         )
 
-    async def async_update_daily() -> MetOfficeData:
+    async def async_update_twice_daily() -> datapoint.Forecast:
         return await hass.async_add_executor_job(
-            fetch_data, connection, site, MODE_DAILY
+            fetch_data, connection, latitude, longitude, "twice-daily"
         )
 
-    metoffice_hourly_coordinator = TimestampDataUpdateCoordinator(
+    async def async_update_hourly() -> datapoint.Forecast:
+        return await hass.async_add_executor_job(
+            fetch_data, connection, latitude, longitude, "hourly"
+        )
+
+    metoffice_daily_coordinator = TimestampDataUpdateCoordinator(
```

Comment:
In a follow-up PR it would be great to see the coordinator moved to a `coordinator.py` file instead.